File: Producto/VeciZervi-BackEnd/backend/src/main/java/com/vecizervi/backend/model/Trabajo.java
package com.vecizervi.backend.model;

import jakarta.persistence.*;
import org.hibernate.annotations.CreationTimestamp;
import java.time.LocalDateTime;

@Entity
@Table(name = "trabajos")
public class Trabajo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_trabajo")
    private Long id;

   
    @ManyToOne
    @JoinColumn(name = "id_cliente", nullable = false)
    private Usuario cliente;

    @ManyToOne
    @JoinColumn(name = "id_categoria", nullable = false)
    private Categoria categoria;

    @Column(nullable = false, length = 100)
    private String titulo;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String descripcion;

    @Column(nullable = false)
    private Double precio;

    @Column
    private Double latitud;

    @Column
    private Double longitud;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EstadoTrabajo estado = EstadoTrabajo.Disponible;

    @CreationTimestamp
    @Column(name = "fecha_publicacion", updatable = false)
    private LocalDateTime fechaPublicacion;

    
    public Trabajo() {
    }
    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }
    public Usuario getCliente() {
        return cliente;
    }
    public void setCliente(Usuario cliente) {
        this.cliente = cliente;
    }
    public Categoria getCategoria() {
        return categoria;
    }
    public void setCategoria(Categoria categoria) {
        this.categoria = categoria;
    }
    public String getTitulo() {
        return titulo;
    }
    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }
    public String getDescripcion() {
        return descripcion;
    }
    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }
    public Double getPrecio() {
        return precio;
    }
    public void setPrecio(Double precio) {
        this.precio = precio;
    }
    public Double getLatitud() {
        return latitud;
    }
    public void setLatitud(Double latitud) {
        this.latitud = latitud;
    }
    public Double getLongitud() {
        return longitud;
    }
    public void setLongitud(Double longitud) {
        this.longitud = longitud;
    }
    public EstadoTrabajo getEstado() {
        return estado;
    }
    public void setEstado(EstadoTrabajo estado) {
        this.estado = estado;
    }
    public LocalDateTime getFechaPublicacion() {
        return fechaPublicacion;
    }
    public void setFechaPublicacion(LocalDateTime fechaPublicacion) {
        this.fechaPublicacion = fechaPublicacion;
    }

    @Column(name = "fecha_finalizacion")
    private LocalDateTime fechaFinalizacion;

    public LocalDateTime getFechaFinalizacion() {
        return fechaFinalizacion;
    }

    public void setFechaFinalizacion(LocalDateTime fechaFinalizacion) {
        this.fechaFinalizacion = fechaFinalizacion;
    }
}

File: Producto/Android Studio/VeciZerviGub/app/src/main/java/com/example/vecizervi/data/models/Trabajo.kt
package com.example.vecizervi.data.models

import com.google.gson.annotations.SerializedName

data class ClienteResumen(
    @SerializedName("idUsuario") val idUsuario: Int? = null,
    @SerializedName("nombres")   val nombres: String = "",
    @SerializedName("apellidos") val apellidos: String = ""
)

data class CategoriaResumen(
    @SerializedName("id") val idCategoria: Int? = null,
    @SerializedName("nombreCategoria") val nombreCategoria: String = ""
)

data class Trabajo(
    @SerializedName("id") val idTrabajo: Long? = null,
    @SerializedName("cliente")            val cliente: ClienteResumen? = null,
    @SerializedName("categoria")          val categoria: CategoriaResumen? = null,
    @SerializedName("titulo")             val titulo: String = "",
    @SerializedName("descripcion")        val descripcion: String = "",
    @SerializedName("comuna")             val comuna: String = "",
    @SerializedName("precio")             val precio: Double = 0.0,
    @SerializedName("estado")             val estado: String = "Disponible",
    @SerializedName("fecha_publicacion")  val fechaPublicacion: String = "",
    @SerializedName("fecha_finalizacion") val fechaFinalizacion: String? = null,
    @SerializedName("latitud")            val latitud: Double? = null,
    @SerializedName("longitud")           val longitud: Double? = null
) {
    val idCliente: Int get() = cliente?.idUsuario ?: 0
}